index.js

```diff
@@ -209,9 +222,38 @@
           }
           return isPropertyDefined('name', config) ? (config.name + '.' + commandName) : commandName;
         };
-
+        
+        // Try to read route configuration and log warning if not possible
+        var routes = {};
+        if (readFromRoute) {
+          if (!$injector.has('$route')) {
+            $log.warn('$route service is not available. Make sure you have included ng-route in your application dependencies.');
+          } else {
+            var $route = $injector.get('$route');
+            routes = $route.routes;
+          }
+        }
         var getUrl = function () {
-          var url = trackUrlParams ? $location.url() : $location.path();
+          // Using ngRoute provided tracking urls
```

Comment:
This section should only happen if `readFromRoute` is enabled, otherwise we should avoid doing this. At present this is implicitly the case due to how `routes` is defined, but explicit would make this clear it is only run in a particular case. Also, one day someone will come along with a use case for routes being defined in another context and this will just get run on every getUrl even if it isn't used.